adr/2025_03_10-suggestions/README.md

```diff
@@ -0,0 +1,178 @@
+
+This would be the most straightforward way to capture changes across versions of a document. The question then becomes how to display those changes within a prosemirror-based editor, which sort of depends on the approach taken for suggestions.
+
+### How should suggestions be stored
+
+Suggestions are different from versioning in that, they are not a history of the document, but a set of pending changes that can be applied to the document. They may be rejected, or may be applied to the document. So, in a sense, they are a patch to the document. But, there are a few different mental models we can use to represent or store these changes to the document:
+
+- [Seeing a suggestion as a _branch of the document history_, and the suggestion as being the diff between the two branches](./suggestions-as-history.md)
+- [Storing document _patches_ and later applying that patch to a document](./suggestions-as-patches.md)
+- [Suggestions as _document content_ which stores the changes within the document](./suggestions-as-content.md)
+
+### AttributionManager
+
+Currently, Y.js has been using a `PermanentUserData` class to keep track and store the `clientID`s that a specific user has used previously. Kevin has some ideas on an improved system for this, which he can elaborate on in detail, later. This will be crucial to how we can attribute users in prior versions of the document.
+
+If I were to share my wish-list for features this should have, it would be able to answer questions like:
+
+- Given a `clientID`, what `userID` made this change?
+- At what time was this `clientID` first used?
+  - This gives approximate times for attributing content to a time, without having to store an additional timestamp per change).
+  - Most people don't need the specific time, they just want to know relative time "Is this days, months or years old?"
+  - Auditing should be done at a different level, and not in a user writable data structure like a YDoc
+
+I'd like to have Kevin give more details on the sort of features/properties we can expect from this. It will be a crucial feature to the implementation, and likely to be worked on first since it will be foundational to how the versioning works.
+
+## Summary
+
+This document was purposefully written from the perspective that Y.js is one of many document storage and synchronization mechanisms. And, we would like to keep our editor de-coupled from these specific implementation details as much as possible to encourage code re-use between different implementations.
+
+## Appendix
+
+This is a collection of scattered thoughts on the _diff_-erent aspects of diffing  (sorry, had to), and their complications.
+
+### Keeping user intent, while diffing
```

Comment:
was thinking about another edge-case earlier today - writing it down here to share and so I don't forget :)

- when a user is in "suggestion" mode, deleting a word, and adding the same word again should probably not result in a deletion and insertion